### PR TITLE
feat: refactor( added Guzzle HTTP client, Unit test and other chores)

### PR DESCRIPTION
**Pull request by ranjanrak**
This PR includes below addition/changes:
1. Added Guzzle HTTP client instead of current Curl.
2. Refactor all required functions to accommodate HTTP client change.  
3. Added Unit test using PHPUnit.
4. Added mock data for the unit tests.

### CODE REVIEW

**Comment by pushpak1300** [comment]
I guess you have added vendor files also

**Comment by ranjanrak** [comment]
Missed adding the Vendor file under .gitignore, last time. Done now :)

**Comment by pushpak1300** [comment]
Still vendor files are visible

**Comment by ranjanrak** [comment]
@pushpak1300 Vendor files had polluted the git tree. The above push should make it somewhat better. You can go through the changes.

**Comment by pushpak1300** [comment]
> If possible, can you comment on feasibility regarding the addition of the WebSocket client as well? Ratchet looks good :)
Yes, I have used Ratchet in the past and it's really good library to work with WebSockets

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{
+    "name": "ranjanrak/phpkiteconnect",
```
Comment (pushpak1300) [inline composer.json:2]: This should be named as `zerodha/phpkiteconnect`

**Comment by pushpak1300** [comment]
Looks Good @ranjanrak 👍🏼

**Comment by ranjanrak** [comment]
Ref [#16](https://github.com/zerodha/phpkiteconnect/pull/16)

**Pull request merged**